velox/expression/tests/ExprTest.cpp

```diff
@@ -3083,3 +3083,17 @@ TEST_F(ExprTest, mapKeysAndValues) {
   std::vector<VectorPtr> result(2);
   ASSERT_NO_THROW(exprSet->eval(rows, context, result));
 }
+
+/// Test recursive constant peeling: in general expression evaluation first,
```

Comment:
nit: How about, "verify that constant peeled result of a child expression is propagated correctly up the expression tree"

Reply:
I'm not happy with the comment I wrote, but the suggested one doesn't seem clearer or accurate to me. The test is orchestrating the following:

- Remove some nulls from the first few rows: 0, 1, 2, 3, 4, -> 2, 3, 4...
- Peel off constant encoding in Expr; save constant index 2.
- Peel off constant encoding in CastExpr; save constant index 0.
- Evaluate cast and produce a result: <undefined>, <undefined>, 5.
- Evaluate expr on top of cast: <undefined>, <undefined>, 15.
- Wrap the result using constant index. The bug was that we would use index 0 and not 2 and end up with <undefined> value.

Reply:
I agree there is a good amount of context to add to understand what this is checking and difficult to capture that in a terse comment.
To clarify, what I meant by my suggested comment was that we ensure that the peeled result is "propagated correctly"(that is, wrapped correctly with the right constantWrapIndex_ before passing it up the expression tree).
Perhaps we can add the terse comment (any one you prefer) at the top of the function and then write out the detailed context above the expression at L3095 for anyone who wants to understand exactly how this expression verifies the same.
What do you think?